applications/FluidDynamicsApplication/custom_elements/distance_smoothing_element.cpp

```diff
@@ -285,12 +285,11 @@ void DistanceSmoothingElement<2>::CalculateLocalSystem(
         }
     }
 
-    // Implementing the boundary condition on distance gradient: can be implemented
-    // by a custom condition for a more general case
+    // Implementing the boundary condition on distance gradient
     const auto& neighbour_elems = this->GetValue(NEIGHBOUR_ELEMENTS);
 
     for (unsigned int i_ne = 0; i_ne < num_faces; i_ne++){
-        if (neighbour_elems[ i_ne ].Id() == this->Id() ){
+        if(this == neighbour_elems(i_ne).get()){
```

Comment:
Why this doesn't work anymore? Ids. must be unique among processors.

Reply:
Accessing Id() without a proxy (using global pointers) is not possible.

Reply:
OK. You're right. Resolving.